rcamp/tests/test_projects_commands.py

```diff
@@ -0,0 +1,259 @@
+            'amount': '50000',
+            'start_date': sdate_tz,
+            'end_date': edate_notice_needed_tz,
+            'expiration_notice_sent': False
+        }
+        alloc_notification_needed = Allocation.objects.create(**alloc_dict_notification_needed)
+
+        # Expired allocations don't need upcoming notification
+        alloc_dict_expired = {
+            'project': self.proj,
+            'allocation_id': 'ucb6_summit2',
+            'amount': '50000',
+            'start_date': sdate_tz,
+            'end_date': edate_expired_tz,
+            'expiration_notice_sent': True
+        }
+        alloc_expired = Allocation.objects.create(**alloc_dict_expired)
+
+        # Notification date in future
+        alloc_dict_no_notification = {
+            'project': self.proj,
+            'allocation_id': 'ucb8_summit3',
+            'amount': '50000',
+            'start_date': sdate_tz,
+            'end_date': edate_far_in_future_tz,
+            'expiration_notice_sent': False
+        }
+        alloc_no_notification = Allocation.objects.create(**alloc_dict_no_notification)
+
+        with mock.patch('django.utils.timezone.now', return_value=current_date_for_test):
+            notification_interval = 20
+            upcoming_expiration_notifier = UpcomingExpirationNotifier(notification_interval)
+            actual_allocations = upcoming_expiration_notifier.get_expiring_allocations_for_interval()
+            expected_query_set = [alloc_notification_needed]
+
+            self.assertEqual(expected_query_set, list(actual_allocations))
+
+    def test_send_upcoming_expiration_notice(self):
+
+        sdate = datetime.datetime(2016,02,02)
+        sdate_tz = pytz.timezone('America/Denver').localize(sdate)
+        current_date_for_test = datetime.datetime(2018,02,01)
+        current_date_for_test_tz = pytz.timezone('America/Denver').localize(current_date_for_test)
+        edate_notice_needed = datetime.datetime(2018,02,21)
+        edate_notice_needed_tz = pytz.timezone('America/Denver').localize(edate_notice_needed)
+
+        alloc_dict_notification_needed = {
+            'project': self.proj,
+            'allocation_id': 'ucb7_summit2',
+            'amount': '50000',
+            'start_date': sdate_tz,
+            'end_date': edate_notice_needed_tz,
+            'expiration_notice_sent': False
+        }
+        alloc_notification_needed = Allocation.objects.create(**alloc_dict_notification_needed)
+
+        with mock.patch('mailer.signals.allocation_expiring.send') as mock_send, \
+                mock.patch('django.utils.timezone.now', return_value=current_date_for_test):
+            notification_interval = 20
+            upcoming_expiration_notifier = UpcomingExpirationNotifier(notification_interval)
+            upcoming_expiration_notifier.send_upcoming_expiration_notices()
+
+            self.assertEqual(mock_send.call_count, 1)
+
+class CommandTestCase(SafeTestCase):
```

Comment:
In coordination with the triplet of suggestions regarding a trailing comma, a test case should be added to validate that a trailing comma is handled correctly.